### PR TITLE
Implemented better sorting of references

### PR DESCRIPTION
**Pull request by samm82**
As per #729, instead of being sorted by UID, references are now sorted with the `compareAuthYearTitle` function, that sorts first by author, then by year, then by title. This is done in a way that isn't hardcoded (via using `Ord` with `Person`, etc.) so that we can easily change the criteria for sorting the references.

### CODE REVIEW

File: code/drasil-lang/Language/Drasil/Reference.hs [hunk 1]
```diff
@@ -271,8 +273,37 @@ instance Referable Contents where
...
+        getStr :: Sentence -> String
+        getStr (S s) = s
+        getStr ((:+:) s1 s2) = getStr s1 ++ getStr s2
+        getStr _ = error "Term is not a string" 
+
 citationsFromBibMap :: BibMap -> [Citation]
-citationsFromBibMap bm = sortBy uidSort citations
+citationsFromBibMap bm = sortBy compareAuthYearTitle citations
   where citations :: [Citation]
         citations = map (\(x,_) -> x) (Map.elems bm)
```
Comment (JacquesCarette) [inline code/drasil-lang/Language/Drasil/Reference.hs]:
```haskell
citations = map fst (Map.elems bm)
```

File: code/drasil-lang/Language/Drasil/Reference.hs [hunk 2]
```diff
@@ -271,8 +273,37 @@ instance Referable Contents where
...
+        isAuthor _          = False 
+
+getYear :: (HasFields c) => c -> Int
+getYear c = maybe (error "No year found") (\(Year x) -> x) (find (isYear) (c ^. getFields))
+  where isYear :: CiteField -> Bool
+        isYear (Year _) = True
+        isYear _        = False
+
+getTitle :: (HasFields c) => c -> String
+getTitle c = getStr $ maybe (error "No title found") (\(Title x) -> x) (find (isTitle) (c ^. getFields))
```
Comment (JacquesCarette) [inline code/drasil-lang/Language/Drasil/Reference.hs]: you don't need () around `isTitle`.

Comment (samm82) [reply code/drasil-lang/Language/Drasil/Reference.hs]: I'm assuming this applies to `isAuthor` and `isYear` as well

Comment (JacquesCarette) [reply code/drasil-lang/Language/Drasil/Reference.hs]: Right, function names do not need to be wrapped in ().

**Comment by samm82** [comment]
@JacquesCarette I made the proposed changes

**Pull request merged**